docs/cunumeric/source/user/usage.rst

```diff
@@ -0,0 +1,127 @@
+Usage
+=====
+
+Drop-in replacement
+-------------------
+
+Using cuNumeric as a replacement for NumPy is easy. Users only need
+to replace:
+
+.. code-block:: python
+
+  import numpy as np
+
+with
+
+.. code-block:: python
+
+  import cunumeric as np
+
+Running cuNumeric programs
+--------------------------
+
+Using ``legate``
+~~~~~~~~~~~~~~~~
+
+The preferred way to run cuNumeric programs is with the ``legate`` driver:
+
+.. code-block:: sh
+
+  legate --cpus 2 --gpus 2 script.py <script options>
+
+All ``legate`` command line configuration options must come *before* the script
+that is be executed. Any options after the script are passed on to the script.
+
+It is also possible to pass the ``legate`` command line configuration options
+using the ``LEGATE_CONFIG`` environment variable:
+
+.. code-block:: sh
+
+  LEGATE_CONFIG="--cpus 2 --gpus 2" legate script.py <script options>
+
+Using standard Python
+~~~~~~~~~~~~~~~~~~~~~
+
+It is also possible to run cuNumeric programs using the standard Python
+interpreter, with some limitations:
+
+.. code-block:: sh
+
+  LEGATE_CONFIG="--cpus 2 --gpus 2" python script.py <script options>
+
+When running programs with this method, configuration options may only be
+passed via the ``LEGATE_CONFIG`` environment variable as shown above.
+
+Additionally, several ``legate`` command line configuration options are not
+available when running programs this way. See the output of ``legate --help``
+for more details.
+
+Multi-node execution
+--------------------
+
+Using ``legate``
+~~~~~~~~~~~~~~~~
+
+Cunumeric programs can be run in parallel by using the ``--nodes`` option to
+the ``legate`` driver, followed by the number of nodes to be used.  Whenever
+the ``--nodes`` option is used, Legate will be launched using ``mpirun``, even
+with ``--nodes 1``.
```

Comment:
I don't think we force "mpirun" anymore, regardless if --nodes is present:

```
~/cunumeric> legate --nodes 1 --verbose a.py

--- Legion Python Configuration ------------------------------------------------

Legate paths:
  legate_dir       : /Users/mpapadakis/legate.core
  legate_build_dir : /Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build
  bind_sh_path     : /Users/mpapadakis/legate.core/bind.sh
  legate_lib_path  : /Users/mpapadakis/legate.core/build/lib

Legion paths:
  legion_bin_path       : /Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build/_deps/legion-build/bin
  legion_lib_path       : /Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build/_deps/legion-build/lib
  realm_defines_h       : /Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build/_deps/legion-build/runtime/realm_defines.h
  legion_defines_h      : /Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build/_deps/legion-build/runtime/legion_defines.h
  legion_spy_py         : /Users/mpapadakis/legion/tools/legion_spy.py
  legion_prof_py        : /Users/mpapadakis/legion/tools/legion_prof.py
  legion_python         : /Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build/_deps/legion-build/bin/legion_python
  legion_module         : /Users/mpapadakis/legion/bindings/python/build/lib
  legion_jupyter_module : /Users/mpapadakis/legion/jupyter_notebook

Versions:
  legate_version : 23.03.00.dev+36.g7e883c5

Command:
  /Users/mpapadakis/legate.core/bind.sh --launcher local -- /Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build/_deps/legion-build/bin/legion_python -ll:py 1 -lg:local 0 -ll:cpu 4 -ll:util 2 -ll:csize 4000 -ll:networks none -level openmp=5 -lg:eager_alloc_percentage 50 -ucx:tls_host '^dc,ud' a.py

Customized Environment:
  DYLD_LIBRARY_PATH=/Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build/_deps/legion-build/lib:/Users/mpapadakis/legate.core/build/lib
  GASNET_MPI_THREAD=MPI_THREAD_MULTIPLE
  LEGATE_MAX_DIM=4
  LEGATE_MAX_FIELDS=256
  NCCL_LAUNCH_MODE=PARALLEL
  PYTHONDONTWRITEBYTECODE=1
  PYTHONPATH=/Users/mpapadakis/legion/bindings/python/build/lib:/Users/mpapadakis/legion/jupyter_notebook:/Users/mpapadakis/legate.core/legate
  REALM_BACKTRACE=1
  REALM_UCP_BOOTSTRAP_PLUGIN=/Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build/_deps/legion-build/lib/realm_ucp_bootstrap_mpi.so
  UCX_CUDA_COPY_MAX_REG_RATIO=1.0
  UCX_IB_RCACHE_PURGE_ON_FORK=n
  UCX_MULTI_LANE_MAX_RATIO=1.0
  UCX_RC_TX_POLL_ALWAYS=y

--------------------------------------------------------------------------------

[0 - 10bb70dc0]    0.000271 {4}{threads}: reservation ('CPU proc 1d00000000000003') cannot be satisfied

~/cunumeric> legate --nodes 2 a.py
ERROR: Could not initialize driver, path config and exception follow:

--- Legion Python Configuration ------------------------------------------------

Legate paths:
  legate_dir       : /Users/mpapadakis/legate.core
  legate_build_dir : /Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build
  bind_sh_path     : /Users/mpapadakis/legate.core/bind.sh
  legate_lib_path  : /Users/mpapadakis/legate.core/build/lib

Legion paths:
  legion_bin_path       : /Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build/_deps/legion-build/bin
  legion_lib_path       : /Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build/_deps/legion-build/lib
  realm_defines_h       : /Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build/_deps/legion-build/runtime/realm_defines.h
  legion_defines_h      : /Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build/_deps/legion-build/runtime/legion_defines.h
  legion_spy_py         : /Users/mpapadakis/legion/tools/legion_spy.py
  legion_prof_py        : /Users/mpapadakis/legion/tools/legion_prof.py
  legion_python         : /Users/mpapadakis/legate.core/_skbuild/macosx--x86_64-3.9/cmake-build/_deps/legion-build/bin/legion_python
  legion_module         : /Users/mpapadakis/legion/bindings/python/build/lib
  legion_jupyter_module : /Users/mpapadakis/legion/jupyter_notebook

Versions:
  legate_version : 23.03.00.dev+36.g7e883c5

--------------------------------------------------------------------------------

Traceback (most recent call last):
  File "/Users/mpapadakis/opt/miniconda3/envs/legate/bin/legate", line 33, in <module>
    sys.exit(load_entry_point('legate.core', 'console_scripts', 'legate')())
  File "/Users/mpapadakis/legate.core/legate/driver/__init__.py", line 37, in main
    return _main(argv)
  File "/Users/mpapadakis/legate.core/legate/driver/main.py", line 73, in legate_main
    driver = prepare_driver(argv, LegateDriver)
  File "/Users/mpapadakis/legate.core/legate/driver/main.py", line 54, in prepare_driver
    raise e
  File "/Users/mpapadakis/legate.core/legate/driver/main.py", line 49, in prepare_driver
    driver = driver_type(config, system)
  File "/Users/mpapadakis/legate.core/legate/driver/driver.py", line 68, in __init__
    self.launcher = Launcher.create(config, system)
  File "/Users/mpapadakis/legate.core/legate/driver/launcher.py", line 118, in create
    return SimpleLauncher(config, system)
  File "/Users/mpapadakis/legate.core/legate/driver/launcher.py", line 295, in __init__
    raise RuntimeError(RANK_ERR_MSG)
RuntimeError: Could not detect rank ID on multi-rank run with no --launcher provided. If you
want Legate to use a launcher, e.g. mpirun, internally (recommended), then you
need to specify which one to use by passing --launcher. Otherwise you need to
invoke the legate script itself through a launcher.
```

Reply:
Legate readme is out of date then

Reply:
Can you suggest some verbiage and I will update in both places

Reply:
Yup, this suggestion would work for the README as well I think https://github.com/nv-legate/cunumeric/pull/805#discussion_r1119401744